apis/listing/listings.py

```diff
@@ -89,29 +105,32 @@ def get_filename(self):
         names = request.form.get('filenames')
         if names:
             filenames = names.split(',')
+            return filenames[0] if filenames[0] else None
+        return None
 
-        return filenames[0] if filenames[0] else None
-
-    def upload_to_s3(self, item, listing_hash):
+    def upload_to_s3(self, listing_hash, location):
         """
         Upload file data to s3, keying it with the listing_hash
         """
         their_md5 = request.form.get('md5_sum')
-        dest = os.path.join('/tmp/uploads/')
-
-        if not os.path.exists(dest):
-            os.makedirs(dest)
-
-        loc = f'{dest}{listing_hash}'
-
-        item.save(loc)
 
-        with open(loc, 'rb') as data:
+        # Read the file the first time to verify the md5 of the uploaded file
```

Comment:
Not thrilled here about reading the same file 3 times, but don't have a clean alternative.
First read: verify the md5sum with what the client provided to verify we received the complete file
Second read: upload to S3
Third read: calculate the keccak so we can set the datahash

Reply:
go ahead an flush it out with the *way too many reads* first. that way we know it works.

once everything is in i'll stare it for a cpl hours, be easier to see then